Add ares configurations

CatTanker opened the pull request:
Aimed at adding Ares Configurations to WAE's YR Config. 

Initial Commit consists of:

- Ares Trigger Events
- Ares Trigger Actions

Which have currently been tested through loading APRA2 campaign maps (which can now be opened without errors) and through comparing how FA2 and WAE save these to the map file. 

ToDo:
- [Scripts](https://ares-developers.github.io/Ares-docs/new/scripts.html)


Presently, I have included a [A] in front of any ares configurations, so that people using vanilla YR can tell that they will not work in the base game.

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+P2Type=HouseType
+
+[EMPulse3]
+Name = Rmoved EM Pulse
```
```suggestion
Name = Removed EM Pulse
```

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+
+[TechnoTypeDoesNotExist]
+Name=TechnoType Does Not Exist
+Description=Triggers when there are no technos of the spercified type on the map
```
```suggestion
Description=Triggers when there are no technos of the specified type on the map

```

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+
+[BuildingDoesNotExist]
+Name=Building Does Not Exist
+Description=No buildings of the specified type (owned by the house of this trigger) exist on the map
```
```suggestion
Description=No buildings of the specified type (owned by the house of this trigger) exist on the map.
```

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+
+[TechnoTypeExists]
+Name=TechnoType Exists
+Description=Triggers when at least this many technos of the specified type exist on the map
```
```suggestion
Description=Triggers when at least this many technos of the specified type exist on the map.
```

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+; ares stuff begins officially
+[EMPulse1]
+Name = Under EM Pulse
+Description = [A] Triggers when the attached object is disabled by EMP
```
```suggestion
Description = [A] Triggers when the attached object is disabled by EMP.
```

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+
+[EMPulse2]
+Name = Under EM Pulse by House
+Description = [A] Triggers when the specified house disables the attached object by EMP. Triggers before 62
```
```suggestion
Description = [A] Triggers when the specified house disables the attached object by EMP. Triggers before 62.
```

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+
+[EMPulse3]
+Name = Rmoved EM Pulse
+Description = [A] Triggers when the attached object is reactivated by a “healing” EMP
```
```suggestion
Description = [A] Triggers when the attached object is reactivated by a “healing” EMP.
```

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+
+[EMPulse4]
+Name = Rmoved EM Pulse by House
+Description = [A] Triggers when the specified house reactivates the attached object by a “healing” EMP. Triggers before 64
```
```suggestion
Description = [A] Triggers when the specified house reactivates the attached object by a “healing” EMP. Triggers before 64.
```

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+Description = [A] Triggers when the attached object is reactivated by a “healing” EMP
+
+[EMPulse4]
+Name = Rmoved EM Pulse by House
```
```suggestion
Name = Removed EM Pulse by House
```

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+
+[Abducted]
+Name = Abducted
+Description = [A] Triggers when the attached object is abducted
```
```suggestion
Description = [A] Triggers when the attached object is abducted.
```

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+Description=Triggers when there are no technos of the spercified type on the map
+P1Type=-2
+P3Type=Techno
+; ares stuff begins officially
```
```suggestion

```

ZivDero replied:
Maybe comments wouldn't hurt, but in a more serious tone
Along the lines of `; ********** ARES ***********`

MortonPL commented on src/TSMapEditor/Config/Events.ini:
```diff
@@ -1,295 +1,463 @@
+Description = [A] Triggers when the specified house has no more buildings with KeepAlive=yes.
+P2Type = HouseType
+
+;end of ares stuff, yay
```
```suggestion
```

MortonPL commented on src/TSMapEditor/Config/Actions.ini:
```diff
@@ -1,619 +1,817 @@
+Name=Jump Camera Home
+Description=Just like hitting the 'H' key.
+
+;begin additional ares stuff
```
```suggestion
```

MortonPL reviewed (changes requested):
Found some typo's and missing punctuation. There are also issues with inconsistent placement of spaces for entries, i.e.:

```ini
[HouseDoesntOwnTechno]
Name =  House Doesn’t Own Techno Type
Description = [A] Triggers while the owning house has fewer than count instances of the specified techno type.
P1Type=-2
P2Type=Number
P3Type=Techno
```

MortonPL commented:
I'm also not sure whether to use [A] or [Ares] or keep the `yr` branch for pure vanilla and have Ares and Phobos elsewhere.

ZivDero commented:
> I'm also not sure whether to use [A] or [Ares] or keep the `yr` branch for pure vanilla and have Ares and Phobos elsewhere.

Desciptions don't need any markings imo, the name can have an [Ares]/[Phobos]
I think having separate configs for Ares/Phobos woud be more of a nuisance.

ZivDero commented:
Also what's up with the diff containing the whole config being rewritten?

CatTanker commented:
![image](https://github.com/Rampastring/WorldAlteringEditor/assets/81521710/737c6157-32af-499b-a291-0cc1688ab545)
My main concern with `[Ares]` is that it takes a lot of space up in the already-existent event/actions list, meaning that in the more severe cases only the first work can be read, which is a limitation if you have 5+ events all beginning with `[Ares] Reverse...`
Ultimately these may have to be resized if we keep with `[Ares]`, and one day `[Phobos]` option.

No clue why the diff is broken.

CatTanker commented:
I have tested each event/action/script from producing a test map inside WAE and comparing the results to E1's FA2sp's understanding. Fixed several errors and updated a few descriptions.  Not much more I can do here.

MortonPL commented:
Can be closed in favor of #202.
